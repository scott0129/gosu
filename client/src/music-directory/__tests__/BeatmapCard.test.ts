import { shallowMount } from '@vue/test-utils';

// 'import' is being buggy with bring in vue components. and it's also old babel, thus 'default'
const BeatmapCard = require('../BeatmapCard.vue').default;

describe('wrapper BeatmapCard', () => {
    const selectionCallback = jest.fn();

    const wrapper = shallowMount(BeatmapCard, {
        propsData: {
            beatmapId: 1,
            version: 'Version 1',
            setId: 22,
            title: 'Fur Elise',
            artist: 'Beethoven',
            previewUrl: 'something.mp3',
            beingPreviewed: false,
        },
    });

    it('renders correct messages', () => {
        expect(wrapper.find('.beatmap-title').text()).toBe('Fur Elise');
        expect(wrapper.find('.beatmap-artist').text()).toBe('Beethoven');
    });

    it('has the expected HTML structure', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('changes html class when being previewed', (done) => {
        wrapper.setProps({
            beingPreviewed: true,
        });
        wrapper.vm.$nextTick(() => {
            wrapper.find('.beatmap-card.selected').exists();
            done()
        })
    });

});
