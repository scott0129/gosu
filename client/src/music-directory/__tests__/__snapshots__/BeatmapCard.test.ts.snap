// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wrapper BeatmapCard has the expected HTML structure 1`] = `
"<div class=\\"beatmap-card\\">
  <h1 class=\\"beatmap-title\\">Fur Elise</h1>
  <h4 class=\\"beatmap-artist\\">Beethoven</h4>
  <div class=\\"audio-player\\"></div>
</div>"
`;
